creativecloud/features/firefly/firefly-masonry.js

```diff
@@ -0,0 +1,202 @@
+  const mediaDiv = [mediaMobile, media];
+  mediaDiv.forEach((div) => {
+    div.appendChild(fireflyPrompt.cloneNode(true));
+    div.appendChild(enticementDiv.cloneNode(true));
+    const promptButton = div.querySelector('.masonry-generate');
+    const promptInput = div.querySelector('.masonry-prompttext');
+    promptButton.addEventListener('click', async (e) => {
+      const userprompt = promptInput?.value;
+      const dall = userprompt === '' ? 'SubmitTextToImage' : 'SubmitTextToImageUserContent';
+      e.target.setAttribute('daa-ll', dall);
+      if (userprompt === '') {
+        window.location.href = allP[4].querySelector('a').href;
+      } else {
+        const { default: signIn } = await import('./firefly-susi.js');
+        signIn(userprompt, 'goToFirefly');
+      }
+    });
+    focusOnInput(null, miloUtil.createTag, promptInput);
+    addKeyEvent(promptInput, promptButton);
+    ic.appendChild(div);
+  });
+}
+
+function processMasonryMedia(gridDiv, miloUtil, allP, mediaDetail) {
+  const lastIndex = mediaDetail.imgSrc.length - 1;
+  const mediaContainer = miloUtil.createTag('div', { class: 'image-container' });
+  const a = miloUtil.createTag('a', { href: `${mediaDetail.href[lastIndex]}` });
+  const img = miloUtil.createTag('img', { src: `${mediaDetail.imgSrc[lastIndex]}`, class: 'prompt-image', alt: `${mediaDetail.alt[lastIndex]}` });
+  const imgPromptContainer = miloUtil.createTag('div', { class: 'image-content' });
+  const imgPrompt = miloUtil.createTag('p', { }, mediaDetail.prompt[lastIndex].trim());
+  const imgHoverIcon = miloUtil.createTag('img', { alt: '', class: 'hoversvg' });
+  imgHoverIcon.src = allP[2].querySelector('a').href;
+  imgPromptContainer.appendChild(imgPrompt);
+  imgPromptContainer.appendChild(imgHoverIcon);
+  a.append(img);
+  a.appendChild(imgPromptContainer);
+  mediaContainer.appendChild(a);
+  handleTouchDevice(mediaContainer, 2000);
+
+  const spanWidth = mediaDetail.spans[lastIndex];
+  mediaContainer.classList.add(`ff-grid-${spanWidth.trim().replace(' ', '-')}`);
+  gridDiv.appendChild(mediaContainer);
+}
+
+function setImgAttrs(a, imagePrompt, src, prompt, href) {
+  const image = new Image();
+  image.src = src;
+  image.onload = () => {
+    a.querySelector('img').src = src;
+    imagePrompt.querySelector('p').innerText = prompt;
+    a.href = href;
+    a.classList.remove('preload');
+  };
+}
+
+function handleAutoCycle(a, mediaDetail, imagePrompt) {
+  const nextIndex = (mediaDetail.index + 1) % mediaDetail.imgSrc.length;
+  const { imgSrc, prompt, href } = mediaDetail;
+  const src = imgSrc[nextIndex];
+  const nextPrompt = prompt[nextIndex];
+  const nextHref = href[nextIndex];
+  setImgAttrs(a, imagePrompt, src, nextPrompt, nextHref);
+  mediaDetail.index = nextIndex;
+  return nextIndex;
+}
+
+function startAutocycle(a, imagePrompt, mediaDetail, interval) {
+  const autocycleInterval = setInterval(() => {
```

Comment:
Can we replace setInterval with setTimeout?

Reply:
Thanks for the review @aishwaryamathuria . I would like to keep it setInterval and change the image attributes after each interval. Please let mw know if  that sounds ok.